### PR TITLE
Refactor: charge_mixing

### PR DESCRIPTION
**Pull request by Qianruipku**
move functions in mixing to cpp files
Before: only one mixing_beta can be used for this mixing
After: We can define any form of mixing_beta function.
Add UTs for double_grid
fix different charge_mixing reults of USPP (NSPIN = 2) because the global varibal in mixing of #3226
Change time out to 1600 s

Update results by #2998 mentioned in #3242
The results changed because:
1. the memory of rho is changed from rho[is][nrxx] to rho[ns*nrxx]
2. ndim = 7 in LCAO is changed to ndim = 8 

### Reminder
- [x] Have you linked an issue with this pull request?
- [x] Have you noticed possible changes of behavior below or in the linked issue?
- [x] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### Linked Issue
Fix #...

### What's changed?
- Example: My changes might affect the performance of the application under certain conditions, and I have tested the impact on various scenarios...

### Any changes of core modules? (ignore if not applicable)
- Example: I have added a new virtual function in the esolver base class in order to ...

### CODE REVIEW

File: source/module_elecstate/module_charge/charge_mixing.cpp [hunk 1]
```diff
@@ -82,54 +92,55 @@ void Charge_Mixing::set_mixing(const std::string& mixing_mode_in,
...
-    GlobalV::ofs_running << "      Autoset mixing_gg0 to " << this->mixing_gg0 << std::endl;
-    if (GlobalV::NSPIN == 1)
-    {
-        GlobalV::MIXING_GG0_MAG = 0.0;
-        GlobalV::ofs_running << "      Autoset mixing_gg0_mag to " << GlobalV::MIXING_GG0_MAG << std::endl;
-    }
-
-    GlobalV::ofs_running << "-------------------------------------" << std::endl;
-}
+// void Charge_Mixing::need_auto_set()
```
Comment (WHUweiqingzhou) [inline source/module_elecstate/module_charge/charge_mixing.cpp:101]: I agree to delete need_auto_set() and auto_set(), which will never be used in the future.

File: source/module_elecstate/module_charge/charge_mixing.h [hunk 2]
```diff
@@ -82,28 +85,29 @@ class Charge_Mixing
                     const double& mixing_beta_in,
                     const int& mixing_ndim_in,
                     const double& mixing_gg0_in,
-                    const bool& mixing_tau_in); // mohan add mixing_gg0_in 2014-09-27
+                    const bool& mixing_tau_in,
+                    const double& mixing_beta_mag_in = 1.6); // mohan add mixing_gg0_in 2014-09-27
```
Comment (WHUweiqingzhou) [inline source/module_elecstate/module_charge/charge_mixing.h]: `mixing_beta_mag` will be set in Default_2() at the very beginning, so it is not necessary to set default value here.

File: source/module_elecstate/module_charge/charge_mixing.h [hunk 3]
```diff
@@ -132,20 +136,36 @@ class Charge_Mixing
...
+    ModulePW::PW_Basis* rhodpw = nullptr; ///< dense grid, same as rhopw for ncpp.
     bool autoset = false;
 
   private:
     double rhog_dot_product(const std::complex<double>* const* const rhog1,
                             const std::complex<double>* const* const rhog2) const;
 
-    void high_freq_mix(std::complex<double>* data, const std::complex<double>* data_save, const int& number) const;
+    /**
+     * @brief divide rho/tau to smooth and high frequency parts
```
Comment (WHUweiqingzhou) [inline source/module_elecstate/module_charge/charge_mixing.h:163]: plz add some explanation about parameters, f. e. `data_d`, `data_s`, etc.

**Review by YuLiu98 — APPROVED** [review]

**Review by WHUweiqingzhou — APPROVED** [review]
Thanks for your effort! This refactor is helpful for the implementation of new mixing method.

File: source/module_elecstate/module_charge/charge_mixing.h (hunk 2 above)
Comment (Qianruipku) [reply source/module_elecstate/module_charge/charge_mixing.h]: Here, I transform GlobalV::MIXING_BETA_MAG to this->mixing_beta. Thus, we can avoid use global variable in charge_mixing.

File: source/module_elecstate/module_charge/charge_mixing.h (hunk 3 above)
Comment (Qianruipku) [reply source/module_elecstate/module_charge/charge_mixing.h:163]: ok

**Comment by hongriTianqi** [comment]
@Qianruipku This PR does not reduce the number of Warnings.

**Pull request merged**